sound/soc/intel/avs/pcm.c

```diff
@@ -1286,16 +1288,20 @@ static const struct snd_soc_dai_driver hda_cpu_dai = {
 		.channels_max	= 8,
 		.rates		= SNDRV_PCM_RATE_8000_192000,
 		.formats	= SNDRV_PCM_FMTBIT_S16_LE |
-				  SNDRV_PCM_FMTBIT_S24_LE |
 				  SNDRV_PCM_FMTBIT_S32_LE,
+		.subformats	= SNDRV_PCM_SUBFMTBIT_MSBITS_20 |
+				  SNDRV_PCM_SUBFMTBIT_MSBITS_24 |
+				  SNDRV_PCM_SUBFMTBIT_MSBITS_32,
 	},
 	.capture = {
 		.channels_min	= 1,
 		.channels_max	= 8,
 		.rates		= SNDRV_PCM_RATE_8000_192000,
 		.formats	= SNDRV_PCM_FMTBIT_S16_LE |
-				  SNDRV_PCM_FMTBIT_S24_LE |
```

Comment:
Would not  just this change could resolve the unsupported S24_LE advertised to userspace?

Reply:
Such change alone is insufficient. It blocks usage of S24_LE yes, but simultaneously does not allow for selecting 20, 24 or 32 as valid bit depth.

Reply:
what is blocking the use of msbits?
Yes, we can only say for example that msbits=32 and have no information that the setup actually supports msbits=24, right?

I mean, should the SNDRV_PCM_SUBFMTBIT_MSBITS_24 be more specific, like SNDRV_PCM_SUBFMTBIT_24_MSBITS_IN_32?
If we really want it to be part of this `struct snd_soc_pcm_stream` or should we have this as refine information for user space that with S32_LE it could use 20, 24 MSB or 24, 32 MSB and let it deal with it?